pysat/instruments/icon_euv.py

```diff
@@ -61,31 +83,35 @@ def init(self):
         modified in-place, as desired.
 
     """
-    logger.info("Mission acknowledgements and data restrictions will be printed " +
-          "here when available.")
+    logger.info(mm_icon.ackn_str)
+    inst.meta.acknowledgements = mm_icon.ackn_str
+    inst.meta.references = ''.join((mm_icon.refs['mission'],
+                                    mm_icon.refs['euv']))
+    if 'keep_original_names' in inst.kwargs.keys():
+        inst.keep_original_names = inst.kwargs['keep_original_names']
```

Comment:
I don't think this should be set here. If a user sees the 'keep_original_names' attribute and changes it there will be no difference to the variables that are loaded.

Reply:
If this code isn't here, I can't initialize an instrument without an error, since the kwargs dict is empty.  Open to alternatives.

Reply:
I missed the change further below. I am behind.

Reply:
Ugh.  grammar. rewriting...

Reply:
ok. I'll be back as soon as I can with something. It is challenging trying to keep up with you and Angeline particularly while juggling other things. :)

Reply:
lol.  I need a break from this code.  I'm starting to cross the different branches in my mind, which may result in some user errors.  I'll refrain from merging until we have a solution in place.  :)

Reply:
OK, so I put in a nominal commit with changes to fuv alone. It works on my system.
While working through it I saw errors in the docstrings related to the keywords. I started a change for FUV.

Looks like I'll need to do at least on more pull for v2.2 to clean up the incorrect docstring statements in a variety of instruments.

Reply:
> OK, so I put in a nominal commit with changes to fuv alone. It works on my system.

Not trying to push off the remaining changes - making sure if it is ok on your side first.

Reply:
OK, looks like I need a longer break since I had evidence already in front of me that it wasn't quite right and yet I didn't recognize it. Cheers.

Reply:
Reporting back that I have a plan on the keywords so that they are always present. I'm going to shift development to another branch so I don't mess with what you have here. On a call now so it will still be a bit.

I think we should also talk about instrument structure at the next pysat meeting.

Reply:
This works for me, but if no keyword is supplied, the remove_preamble routine is not run.

Reply:
Yeah, I'm going to collect keyword default values when I collect the keywords during pysat.Instrument instantiation. Keywords will always be in .kwargs here soon.